src/Sarif/Visitors/GitHubIngestionVisitor.cs

```diff
@@ -144,6 +119,8 @@
 
         public override Result VisitResult(Result node)
         {
+            this.ruleIndex = node.RuleIndex;
+
             if (node.Fingerprints != null)
             {
                 // GitHub appears to require that fingerprints be emitted to the
```

Comment:
>VisitMessage [](http://example.com/codeflow?start=32&length=12)

We have a message string look-up helper that you should use. As Kalle notes, this is a pretty complex operation.

Reply:
OK, nope. It appears that we don't. So the correct course here is to please create a common helper for both this code and the insert optional data visitor to call. Second, create an issue for the problem that Kalle notes: 'Message creation logic does not properly search tool extensions for format strings.'

Reply:
got it, will work on this.

Reply:
done both add helper method and create an issue
[https://github.com/microsoft/sarif-sdk/issues/2582](https://github.com/microsoft/sarif-sdk/issues/2582)